lib/routes/validators.js

```diff
@@ -170,6 +170,13 @@ function isValidUrl(url, hostnameRegex) {
   // The only trick here is that `new URL()` will add a trailing
   // slash if there's no path component, which is why we also
   // compare to `origin` below.
+
+  // if the original url has a :443 in the url we need to strip it for the check below
+  // to follow the `new URL` behaviour
+  if (/^https:\/\//.test(url) && /:443/.test(url)) {
```

Comment:
I was testing this and think it needs to be:

```js
if (/^https:\/\//.test(url) && /:443($|\/[^\d])/.test(url)) {
```

That avoids `:4430` from having it's port replaced. It says, replace `:443 if it's at the end of the URL, or if the character following `:443` is not a digit

Reply:
that would be a good improvement 👍

Reply:
Is there a reason not to use `url.parse` instead of regular expressions?

Reply:
Actually, widening the view, `parsed.port` is already available (`null` if no explicit port number).

Reply:
@jrgm we already use `new URL` above